### PR TITLE
Tidy up project toml files

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**
1. Removes non-mandatory attributes and text entries.

1. Bumps all depepdencies except:
    * jsonwebtoken (as @funbringer was willing to bump it later)
    * rust-tls + webpki + rustls-split crate circle. Agreed to remove rustls-split first from the workspace separately.

1. Removes parking_lot from the project due to https://github.com/zenithdb/zenith/issues/956

1. Moves `routerify` behind `zenith_utils` to the rest of the generic http server code

### CODE REVIEW

File: zenith/src/main.rs [hunk 1]
```diff
@@ -445,10 +441,8 @@ fn handle_tenant(tenant_match: &ArgMatches, env: &local_env::LocalEnv) -> Result
             pageserver.tenant_create(tenantid)?;
             println!("tenant successfully created on the pageserver");
         }
-
-        (sub_name, _) => {
-            bail!("Unexpected tenant subcommand '{}'", sub_name)
-        }
+        Some((sub_name, _)) => bail!("Unexpected tenant subcommand '{}'", sub_name),
+        None => bail!("No tenant subcommand found"),
```
Comment (hlinnaka) [inline zenith/src/main.rs:445]: The error messages on missing subcommand are inconsistent:

No tenant subcommand found
no pg subcommand provided
No pageserver subcommand given
no safekeeper subcommand provided

Let's standardize on some variant. I like the "no X subcommand provided" the best. Not sure if it should be capitalized or not, but let's try to be consistent on that too.

File: zenith_utils/src/http/mod.rs [hunk 2]
```diff
@@ -2,3 +2,7 @@ pub mod endpoint;
 pub mod error;
 pub mod json;
 pub mod request;
+
+/// Current fast way to applly simple http routing in varuious Zenith binaries.
+/// Reexported for sake of uniform approach, that could be later replaced with better alternatives, if needed.
+pub use routerify::{ext::RequestExt, RouterBuilder};
```
Comment (hlinnaka) [inline zenith_utils/src/http/mod.rs:8]:
```suggestion

/// Current fast way to apply simple http routing in various Zenith binaries.
/// Re-exported for sake of uniform approach, that could be later replaced with better alternatives, if needed.
pub use routerify::{ext::RequestExt, RouterBuilder};
```
I don't understand what this is all about though. Why is the re-export needed?

File: zenith_utils/src/auth.rs [hunk 3]
```diff
@@ -113,6 +112,14 @@ impl JwtAuth {
...
 }
 
+impl std::fmt::Debug for JwtAuth {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("JwtAuth")
+            .field("validation", &self.validation)
+            .finish()
+    }
+}
+
```
Comment (hlinnaka) [inline zenith_utils/src/auth.rs:122]: If I understand correctly, the idea is to hide the 'decoding_key' from debugging format. That's sensible. This change is currently part of the "bump pageversion dependencies" commit, was this required by changes in dependencies? This seems like a good idea on its own, so maybe extract it to a separate commit.

**Review by hlinnaka — APPROVED** [review]
Left some minor comments, +1 in general.

File: pageserver/src/virtual_file.rs [hunk 4]
```diff
@@ -332,8 +332,10 @@ impl VirtualFile {
         // TODO: We could downgrade the locks to read mode before calling
         // 'func', to allow a little bit more concurrency, but the standard
         // library RwLock doesn't allow downgrading without releasing the lock,
-        // and that doesn't seem worth the trouble. (parking_lot RwLock would
-        // allow it)
+        // and that doesn't seem worth the trouble.
+        //
+        // XXX `parking_lot::RwLock` can enable such downgrades, yet its implemenation is fair and
+        // may deadlock on subsequent read calls, not all code places would benefit from such benaviour
```
[the remark this answers is not in the document]
Comment (SomeoneToIgnore) [reply pageserver/src/virtual_file.rs:338]: I'm afraid, `EphemeralFile` and `VirtualFile` are actually the cause of deadlocks in our tests right now, if I just replace the `RwLock` import: that was the case last time I've checked at least, if I've understood the traces right.
But we can rewrite the code and still use the downgradable lock later.

Either way, elaborated a bit.

File: zenith/src/main.rs [hunk 5]
```diff
@@ -67,134 +67,128 @@ struct BranchTreeEl {
...
-        .help("Node name")
-        .required(true);
-
-    let safekeeper_node_arg = Arg::with_name("node")
-        .index(1)
-        .help("Node name")
-        .required(false);
+    let pg_node_arg = Arg::new("node").index(1).help("Node name").required(true);
+
+    let safekeeper_node_arg = Arg::new("node").index(1).help("Node name").required(false);
```
[the remark this answers is not in the document]
Comment (SomeoneToIgnore) [reply zenith/src/main.rs:72]: Yes, but we can use `#[rustfmt::skip]` to enforce such things, so updated.

File: pageserver/src/virtual_file.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (hlinnaka) [reply pageserver/src/virtual_file.rs:338]: Oh, ok then

File: zenith_utils/src/http/mod.rs (hunk 2 above)
Comment (SomeoneToIgnore) [reply zenith_utils/src/http/mod.rs:8]: @LizardWizzard  and @funbringer most probably can elaborate better than me, since I've followed their advice here :) 

In general, the idea is to put all that we use in http server set-up (routing, tokens, etc.) behind a single crate and use it in all binaries.
This way we get:
* all http-related things in one place (`zenith_utils::http::*`)
* possible ease the dependency upgrades later (well, not for this particular case since we reexport, but in theory)
* same code for all places that can be later replaced with a "more decent" http/routing/whatever library

File: zenith_utils/src/auth.rs (hunk 3 above)
Comment (SomeoneToIgnore) [reply zenith_utils/src/auth.rs:122]: Sure.

Initially that came as a preparation for new `jsonwebtoken` library version update, that turned the `DecodingKey` into `!Debug`, but indeed this is not obvious, extracted.

File: zenith_utils/src/http/mod.rs (hunk 2 above)
Comment (hlinnaka) [reply zenith_utils/src/http/mod.rs]: Ok, make sense!

**Review by hlinnaka — APPROVED** [review]

**Review by LizardWizzard — APPROVED** [review]
Nice! Thanks! I'll mention jsonwebtoken update somewhere near enabling interservice authentication

**Comment by funbringer** [comment]
I'm late yet again, but I can't help but to notice that in my opinion commit https://github.com/zenithdb/zenith/commit/c5b5905ed36490bbaaca4081236b7f205afdb68c does more harm than good. The [proposed reasoning](https://github.com/zenithdb/zenith/pull/1127#issuecomment-1015323930) mostly covers **rwlocks**, however, the patch itself removes **mutex** from a place where it fits quite well (come on, we're doing a bunch of inserts to a hash table, what could possibly go wrong?). The fairness guarantees are also beneficial here; ideally we wouldn't want to starve any threads, especially if we could get this behavior for free. Finally, the commit message is misleading: our dependencies still rely upon parking_lot, so we didn't actually remove it from our workspace.

That is to say, unless there's some serious concern regarding parking_lot that hasn't been brought up yet, I'm going to use it in proxy as necessary.
